tests/lib/cputemp/test_ble_tools.py

```diff
@@ -26,6 +27,53 @@ def test_find_adapter(self, mock_dbus_interface):
         bus.get_object.assert_called()
         self.assertEqual(result, None)
 
+    @patch('dbus.SystemBus')
+    @patch('dbus.Interface')
+    def test_no_adapter(self, mocked_dbus_interface, mocked_system_bus):
+        def system_bus_side_effect(*args, **kwargs):
+            def get_object_side_effect(bus_name, object_path):
+                # When there is no BLE adapter, object_path became None
+                if not object_path:
+                    raise TypeError()
+                return MagicMock()
+
+            mock_dbus = MagicMock()
+            mock_dbus.get_object = MagicMock()
+            mock_dbus.get_object.side_effect = get_object_side_effect
+            return mock_dbus
+
+        mocked_system_bus.side_effect = system_bus_side_effect
+
+        # Test BluetoothServicesProcessor
```

Comment:
A few ways to improve these tests:

- Test the code that is part of this PR. The new code adds logging but you are testing exception throwing.
- Only test one thing per test. This combines Advertisement and BluetoothServicesProcessor testing.
- Better pattern for asserting exceptions: https://github.com/NebraLtd/hm-diag/blob/4ab18f41e4f49ccad14f57b2e894e95cac3b7b78/hw_diag/tests/test_blockchain_helium_height.py#L46

Reply:
I pushed the patch.